frontend/javascripts/admin/project/transfer_all_tasks_modal.tsx

```diff
@@ -115,60 +102,41 @@ class TransferAllTasksModal extends React.PureComponent<Props, State> {
     );
   }
 
-  handleSelectChange = (userId: string) => {
-    this.setState((prevState) => {
-      const selectedUser = prevState.users.find((user) => user.id === userId);
-      return {
-        selectedUser,
-      };
-    });
-  };
-
-  render() {
-    const project = this.props.project;
+  function handleSelectChange(userId: string) {
+    setSelectedUser(users.find((user) => user.id === userId));
+  }
 
-    if (!project) {
-      return (
-        <Modal title="Error" open onOk={this.props.onCancel} onCancel={this.props.onCancel}>
-          <p>{messages["project.none_selected"]}</p>
-        </Modal>
-      );
-    } else {
-      const title = `All users with active tasks for ${project.name}`;
-      return (
-        <Modal
-          title={title}
-          open
-          onCancel={this.props.onCancel}
-          // @ts-expect-error ts-migrate(2322) FIXME: Type '{ children: (string | Element)[]; title: str... Remove this comment to see the full error message
-          pagination="false"
-          footer={
-            <div>
-              <Button
-                type="primary"
-                disabled={!this.state.selectedUser}
-                onClick={this.transferAllActiveTasks}
-              >
-                Transfer all tasks
-              </Button>
-              <Button onClick={this.props.onCancel}>Close</Button>
-            </div>
-          }
-        >
-          <div>
-            {this.state.isLoading ? <Spin size="large" /> : this.renderTableContent()}
-            <br />
-            <br />
-          </div>
-          Select a user to transfer the tasks to:
-          <div className="control-group">
-            <div className="form-group">
-              <UserSelectionComponent handleSelection={this.handleSelectChange} />
-            </div>
+  if (!project) {
+    return (
+      <Modal title="Error" open onOk={onCancel} onCancel={onCancel}>
+        <p>{messages["project.none_selected"]}</p>
+      </Modal>
+    );
+  } else {
+    const title = `All users with active tasks for ${project.name}`;
+    return (
+      <Modal
+        title={title}
+        open
+        onCancel={onCancel}
+        onOk={transferAllActiveTasks}
+        okText="Transfer all tasks"
```

Comment:
Use the `onOk`, `okText` `okButtonProps` props here instead of `footer`. This keeps the default styling of antd, including proper margins.